src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala

```diff
@@ -486,6 +489,132 @@ trait BronzeTransforms extends SparkSessionWrapper {
+      }
+    }
+
+    var clusterListIDs = Array[String]()
+
+    try {
+      val clusterListDF = workspace.getClustersDF(deriveApiTempDir(config.tempWorkingDir,apiEndpointTempDir,pipelineSnapTS))
+      if (!clusterListDF.isEmpty) {
+        clusterListIDs = clusterListDF.select("cluster_id").distinct().collect().map(x => x(0).toString)
+      }
+    } catch {
+      case e: Exception => {
+        val message = s"No new clusters retrieved from the cluster/list API"
+        logger.log(Level.WARN, message)
+      }
+    }
+
+    val errorMsg = "No Data is present for cluster snapshot"
+
+    val allClusterIDs = (clusterIDs ++ jobClusterIDs ++ clusterListIDs).distinct
+
+    if (allClusterIDs.isEmpty) throw new NoNewDataException(s"No clusters could be found with new events. Please " +
+      s"validate your audit log input and clusters_snapshot_bronze tables to ensure data is flowing to them " +
+      s"properly. Skipping!", Level.ERROR)
+
+    logger.log(Level.INFO, f"Retrieving cluster information for ${clusterIDs.length} clusters")
+
+    val tmpClusterSnapshotSuccessPath = s"${config.tempWorkingDir}/${apiEndpointTempDir}/success_" + pipelineSnapTS.asUnixTimeMilli
+    val tmpClusterSnapshotErrorPath = s"${config.tempWorkingDir}/${apiEndpointTempDir}/error_" + pipelineSnapTS.asUnixTimeMilli
+
+    landClusterSnapshot(allClusterIDs, pipelineSnapTS.asUnixTimeMilli, tmpClusterSnapshotSuccessPath,
+      tmpClusterSnapshotErrorPath, config)
+
+    logger.log(Level.INFO, " cluster snapshot landing completed")
+
+    if (Helpers.pathExists(tmpClusterSnapshotErrorPath)) {
+      persistErrors(
+        spark.read.json(tmpClusterSnapshotErrorPath),
+        database,
+        clusterSnapshotErrorsTarget,
+        pipelineSnapTS,
+        config.organizationId
+      )
+    }
+
+    if (Helpers.pathExists(tmpClusterSnapshotSuccessPath)) {
+      try {
+        val rawDF = deriveRawApiResponseDF(spark.read.json(tmpClusterSnapshotSuccessPath))
+        if (rawDF.columns.contains("cluster_id")) {
+          val outputDF = SchemaScrubber.scrubSchema(rawDF)
+          val finalDF = outputDF.withColumn("default_tags", SchemaTools.structToMap(outputDF, "default_tags"))
+            .withColumn("custom_tags", SchemaTools.structToMap(outputDF, "custom_tags"))
+            .withColumn("spark_conf", SchemaTools.structToMap(outputDF, "spark_conf"))
+            .withColumn("spark_env_vars", SchemaTools.structToMap(outputDF, "spark_env_vars"))
+            .withColumn(s"aws_attributes", SchemaTools.structToMap(outputDF, s"aws_attributes"))
+            .withColumn(s"azure_attributes", SchemaTools.structToMap(outputDF, s"azure_attributes"))
+            .withColumn(s"gcp_attributes", SchemaTools.structToMap(outputDF, s"gcp_attributes"))
+            .withColumn("organization_id", lit(config.organizationId))
+            .verifyMinimumSchema(clusterSnapMinimumSchema)
+          finalDF
+
+        } else {
+          val msg = "No New Records for Snapshot"
```

Comment:
You have the empty error message in line 539, you should use that in all 3 cases